### PR TITLE
add validation that the commit by parameter matches the git commitof the project

### PR DESCRIPTION
**Pull request by pedrocodacy**

### CODE REVIEW

File: cli/src/main/scala/com/codacy/analysis/cli/Main.scala [hunk 1]
```diff
@@ -64,14 +64,21 @@ class MainImpl extends CLIApp {
...
 
-  private def validate(configuration: CLIConfiguration): Either[CLIError, Unit] = {
+  private def validate(analyse: Analyse, configuration: CLIConfiguration): Either[CLIError, Unit] = {
+    for {
+      _ <- validateGitStatus(configuration.analysis.projectDirectory, configuration.upload.upload)
+      _ <- validateCommitUuid(configuration.analysis.projectDirectory, analyse.commitUuid)
+    } yield ()
+  }
+
+  private def validateGitStatus(projectDirectory: File, upload: Boolean): Either[CLIError, Unit] = {
```
Comment (xplosunn) [inline cli/src/main/scala/com/codacy/analysis/cli/Main.scala]: Maybe rename to "validateGitTrackedFiles" or "allFilesTrackedByGIt" ?

File: cli/src/main/scala/com/codacy/analysis/cli/Main.scala [hunk 2]
```diff
@@ -84,6 +91,22 @@ class MainImpl extends CLIApp {
     }).getOrElse(Right(()))
   }
 
+  private def validateCommitUuid(projectDirectory: File, commitUuidOpt: Option[Commit.Uuid]): Either[CLIError, Unit] = {
+    (for {
+      repo <- Git.repository(projectDirectory).toOption
+      gitCommit <- repo.latestCommit.toOption
+      paramCommit <- commitUuidOpt
```
Comment (xplosunn) [inline cli/src/main/scala/com/codacy/analysis/cli/Main.scala]: "paramCommit" =>"paramCommitUuid"

**Comment by pedrocodacy** [comment]
This PR should not be merged until @rtfpessoa finishes the merge of #108

**Comment by pedrocodacy** [comment]
#108 is done, this is now ready for merge

File: cli/src/main/scala/com/codacy/analysis/cli/Main.scala [hunk 3]
```diff
@@ -84,6 +91,22 @@ class MainImpl extends CLIApp {
...
+    } yield {
+      if (gitCommit.commitUuid != paramCommitUuid) {
+        val error = CLIError.CommitUuidsDoNotMatch(paramCommitUuid, gitCommit.commitUuid)
+        logger.error(error.message)
+        Left(error)
+      } else {
+        Right(())
+      }
+    }).getOrElse(Right(()))
+  }
```
Comment (xplosunn) [inline cli/src/main/scala/com/codacy/analysis/cli/Main.scala]: The "if" can be moved into the for comprehension so we have a single Right case. Check if you think it's cleaner or harder to understand that way.

**Pull request merged**